server/methods.coffee

```diff
@@ -128,192 +161,19 @@ Meteor.methods
-        # Detect whether count is cumulative
-        if 'incremental' in attributes
-          incident.dateRange.cumulative = false
-        else if 'cumulative' in attributes
-          incident.dateRange.cumulative = true
-        else if incident.dateRange.type == 'day' and count > 300
-          incident.dateRange.cumulative = true
-        suspectedAttributes = _.intersection([
-          'approximate', 'average', 'suspected'
-        ], attributes)
-        if suspectedAttributes.length > 0
-          incident.status = 'suspected'
-        incident.url = source.url
-        # The disease field is set to the last disease mentioned.
-        diseaseTerritory.annotations.forEach (annotation)->
-          incident.resolvedDisease =
-            id: annotation.resolutions[0].uri
-            text: annotation.resolutions[0].label
-        incident.suggestedFields = _.intersection(
-          Object.keys(incident),
-          [
-            'resolvedDisease'
-            'cases'
-            'deaths'
-            'dateRange'
-            'status'
-            if incident.locations.length then 'locations'
-          ]
-        )
-        if incident.dateRange?.cumulative
-          incident.suggestedFields.push('cumulative')
-
-        annotations =
-          case: [
-            textOffsets: incident.countAnnotation.textOffsets[0]
-            text: incident.countAnnotation.text
-          ]
-        if locationTerritory.annotations.length
-          annotations.location =
-            locationTerritory.annotations.map (a) -> textOffsets: a.textOffsets[0]
-        if dateTerritory.annotations.length
-          annotations.date =
-            dateTerritory.annotations.map (a) -> textOffsets: a.textOffsets[0]
-        if diseaseTerritory.annotations.length
-          annotations.disease =
-            diseaseTerritory.annotations.map (a) -> textOffsets: a.textOffsets[0]
-        incident.annotations = annotations
-
-        incidents.push(incident)
-
-      incidentData =
-        content: enhancements.source.cleanContent.content
-        incidents: incidents
-
-      if addToCollection
-        _incidents = incidents.map (incident) ->
-          _incident = _.pick(incident, incidentReportSchema.objectKeys())
-        Meteor.call 'addIncidentReports', _incidents, false, (error, result) ->
-          return incidentData
-
-      else
-        return incidentData
+  addSourceIncidentReportsToCollection: (source, options) ->
+    { acceptByDefault } = options
+    enhancements = source.enhancements
+    check enhancements, Object
+    options.url = cleanUrl(source.url)
+    options.publishDate = source.publishDate
+    incidents = createIncidentReportsFromEnhancements(enhancements, options)
+    incidents = incidents.map (incident) ->
+      incident = _.pick(incident, incidentReportSchema.objectKeys())
+    # Remove prior unaccepted incident reports for the article
+    Incidents.remove(
+      url: $regex: regexEscape(options.url) + "$"
```

Comment:
Could someone remove unexpected incidents by passing in a URL other than what is expected?  For example, if `promedmail.org` was passed in, would this just remove all unaccepted, autogenerated incidents?